interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/TypeInfos.scala

```diff
@@ -87,13 +97,12 @@ object TypeInfos {
         typeFunction(arguments).leftMap(_.map(SpelExpressionParseErrorConverter(this, arguments).convert(_)))
         }
         .map{res =>
-          if (!res.canBeSubclassOf(staticResult)) throw new AssertionError("Generic function returned type that does not match static parameters.")
-            res
+          if (!staticInfo.signatures.map(_.result).exists(res.canBeSubclassOf))
```

Comment:
IMO better would be to check against return type returned by `staticInfo.computeResultType`. Won't be?